auth/service.go

```diff
@@ -22,26 +22,45 @@ import (
 	"net/http"
 )
 
-type Service struct {
-	Config        *configuration.Data
-	ClientOptions []configuration.HTTPClientOption
-	SaToken       string
+type Service interface {
+	GetUser(ctx context.Context) (*User, error)
+	GetAuthURL() string
+	NewSaClient() (*authclient.Client, error)
+	ResolveUserToken(ctx context.Context, target, userToken string) (user, accessToken string, err error)
+	ResolveSaToken(ctx context.Context, target string) (username, accessToken string, err error)
+	GetClientOptions() []configuration.HTTPClientOption
+	GetPublicKeys() ([]*rsa.PublicKey, error)
```

Comment:
We already have Auth service in common. We need to add more stuff like this to common and reuse it here.
It's related to https://github.com/fabric8-services/fabric8-cluster/issues/8 (@dipak-pawar wink wink)